### PR TITLE
[top] Add partial SRAM scrambling device to top-level

### PR DESCRIPTION
**Pull request by msfschaffner**
This is in preparation to for instantiating the `sram_ctrl` module, and instantiates the scrambling device in the top level (without CSRs and support for key request). The main goal here is to get this up and running with the default all-zero key such that we have this block covered in our regressions.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Comment by msfschaffner** [comment]
Hm... it looks like the memory scrambler works in principle, but it increases the simulation runtime, thereby leading to CI timeouts. Any ideas how we could mitigate this?

**Comment by tjaychen** [comment]
eh? it increases it by that much?
could there maybe be something else going on?\

**Comment by msfschaffner** [comment]
I'll dig in some more next week to see what is going on.

**Comment by tjaychen** [comment]
does the test finish locally for you?

File: hw/top_earlgrey/top_earlgrey.core [hunk 1]
```diff
@@ -26,6 +26,7 @@ filesets:
       - lowrisc:ip:kmac
       - lowrisc:ip:otbn
       - lowrisc:prim:ram_1p_adv
```
Comment (imphil) [inline hw/top_earlgrey/top_earlgrey.core:28]: You can remove this dependency, I think.

**Review by imphil — COMMENTED** [review]
Looking at the test results, it could well be that one of the tests is simply failing. It then times out due to a bug in the test setup, which will be fixed as soon as https://github.com/lowRISC/opentitan/pull/4095 is merged.

**Comment by imphil** [comment]
You also need to update the memory path at `hw/top_earlgrey/top_earlgrey_verilator.cc`.

File: hw/top_earlgrey/top_earlgrey.core (hunk 1 above)
Comment (msfschaffner) [reply hw/top_earlgrey/top_earlgrey.core]: yes this is not needed anymore - removed.

**Comment by msfschaffner** [comment]
> You also need to update the memory path at `hw/top_earlgrey/top_earlgrey_verilator.cc`.

oh yeah that's right. fixed!

**Review by eunchan — APPROVED** [review]

**Comment by msfschaffner** [comment]
> Looking at the test results, it could well be that one of the tests is simply failing. It then times out due to a bug in the test setup, which will be fixed as soon as #4095 is merged.

Indeed that may be the case as well. I have not had time to dig into this yet. Will wait until #4095 is in, rebase and give it another look.

**Comment by imphil** [comment]
@msfschaffner  #4095 is in, you might want to rebase to see what the test tells you.

**Comment by msfschaffner** [comment]
Ok finally had time to look at this. There was a sequencing bug in access sequences with alternating R/W ops. This should be fixed now.

**Pull request merged**